### ISSUE TITLE
S2259 False positive Null pointers should not be dereferenced

### ISSUE THREAD

**Issue by jorisvergeer** [SonarSource/sonar-dotnet#2593]
### Description

```csharp
//Assume obj is a instance of a class with the property Id and a property Name
var objId = obj?.id;
if(objId.HasValue)
{
  ...
  var objName = obj.Name; // <-- Issue raised here
  ...
}
```

If `objId` has a value, then `obj` could hot have been `null`. Therefore when its Name property is accessed _without_ a null propagation operator, then it is not possible that `obj.Name` could cause errors.

### Expected behavior

No issue

### Actual behavior

S2259 is raised

### Related information

sonarcloud.io as of 20-08-2019
SonarScanner for MSBuild 4.6.2
SonarQube Scanner 3.3.0.1492
SonarQube server 8.0.0

**Comment by andrei-epure-sonarsource** [comment]
hey @jorisvergeer , thanks for the repro - I cannot repro using the latest version ([7.16](https://www.nuget.org/packages/SonarAnalyzer.CSharp/7.16.0.8981)) and the below snippet

```
    public class ReproFor2593
    {
        public int id;
        public string name;

        public void Repro(ReproFor2593 obj)
        {
            obj = null;
            var objId = obj?.id;
            if (objId.HasValue)
            {
                var objName = obj.name; // no issue here
            }
        }
    }
``` 

could you please update to the latest version of the plugin? you can see we fixed a lot of FPs in [7.16 release notes](https://github.com/SonarSource/sonar-dotnet/releases/tag/7.16.0.8981) so I expect this got fixed then.

also, I expect you don't have `SonarQube server 8.0.0` as the latest version is `7.9.1`

thanks
